### PR TITLE
Fix #12, add bp coverage test framework

### LINKED ISSUE #12: Add coverage test framework for BP App
All CFS software needs to have coverage testing, but this is not yet implemented for BP.  

Initially, the first enhancement would be to add the coverage framework only, with empty tests, such that the coverage test shows lines and functions NOT being covered (as opposed to showing nothing at all).

After this is in place, test cases can be gradually added to achieve the desired coverage (ideally 100% of lines, functions, and branches)

### PR DESCRIPTION
**Pull request by jphickey**
Adds the initial framework for BP coverage test.
Currently no test cases, so coverage level is low, but it shows in the report.

Fixes #12

### CODE REVIEW

**Review by dzbaker — APPROVED** [review]

**Comment by jphickey** [comment]
This one cannot be merged until nasa/bplib#174 is also merged, as it is a dependency.  So holding off until that is done.

**Pull request merged**
